code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -550,11 +574,9 @@
 /obj/structure/carp_rift/Destroy()
 	STOP_PROCESSING(SSobj, src)
 	if(time_charged != max_charge + 1)
-		to_chat(dragon, "<span class='boldwarning'>The rift has been destroyed!  You have failed, and find yourself brought down by the weight of your failure.</span>")
-		dragon.set_varspeed(5)
-		dragon.tiredness_mult = 5
-		dragon.destroy_rifts()
-		playsound(src, 'sound/vehicles/rocketlaunch.ogg', 100, TRUE)
+		dragon?.destroy_rifts()
+		if(dragon)
+			to_chat(dragon, "<span class='boldwarning'>A rift has been destroyed!  You have failed, and find yourself weakened.</span>")
```

Comment:
```suggestion
			to_chat(dragon, "<span class='boldwarning'>A rift has been destroyed! You have failed, and find yourself weakened.</span>")
```

The `if` is also unnecessary, you can `to_chat` null and it's just a no-op, but doesn't matter.